front/src/modules/timesStops/TimesStopsOutput.tsx

```diff
@@ -31,17 +35,34 @@ const TimesStopsOutput = ({
   simulatedPathItemTimes,
   operationalPointsOnPath,
 }: TimesStopsOutputProps) => {
-  const rows = useOutputTableData(
+  const useNewTimesStopsTable = useSelector(getUseNewTimesStopsTable);
+
+  // Only call the hook that corresponds to the active table to avoid unnecessary computation
+  const legacyRows = useOutputTableData(
     infraId,
     isValid,
-    selectedTrain,
-    simulatedTrain,
-    simulatedPathItemTimes,
-    operationalPointsOnPath
+    useNewTimesStopsTable ? undefined : selectedTrain,
+    useNewTimesStopsTable ? undefined : simulatedTrain,
+    useNewTimesStopsTable ? undefined : simulatedPathItemTimes,
+    useNewTimesStopsTable ? undefined : operationalPointsOnPath
   );
+
+  const newRows = useTimesStopsTableData(
+    infraId,
+    isValid,
+    useNewTimesStopsTable ? selectedTrain : undefined,
+    useNewTimesStopsTable ? simulatedTrain : undefined,
+    useNewTimesStopsTable ? simulatedPathItemTimes : undefined,
+    useNewTimesStopsTable ? operationalPointsOnPath : undefined
+  );
+
+  if (useNewTimesStopsTable) {
+    return <TimesStopsTable rows={newRows} dataIsLoading={!selectedTrain} />;
```

Comment:
Hum I'm wondering if the correct condition should be `newRows.length === 0` for `dataIsLoading`
```suggestion
    return <TimesStopsTable rows={newRows} dataIsLoading={!selectedTrain || newRows.length === 0} />;
```